gnome-initial-setup/gis-driver.c

```diff
@@ -288,6 +294,118 @@ gis_driver_get_mode (GisDriver *driver)
 }
 
 gboolean
+gis_driver_is_in_demo_mode (GisDriver *driver)
+{
+  GisDriverPrivate *priv = gis_driver_get_instance_private (driver);
+  return priv->is_in_demo_mode;
+}
+
+#define DEMO_ACCOUNT_USERNAME "demo-guest"
+#define DEMO_ACCOUNT_FULLNAME "Demo"
+
+static void
+handle_demo_mode_error (GError *error)
+{
+  GtkWidget *dialog;
+
+  g_warning ("%s", error->message);
+  dialog = gtk_message_dialog_new (NULL,
+                                   GTK_DIALOG_DESTROY_WITH_PARENT,
+                                   GTK_MESSAGE_ERROR,
+                                   GTK_BUTTONS_CLOSE,
+                                   "Failed to enter demo mode: %s",
+                                   error->message);
+  gtk_dialog_run (GTK_DIALOG (dialog));
+  gtk_widget_destroy (dialog);
+  g_error_free (error);
+}
+
+G_DEFINE_AUTOPTR_CLEANUP_FUNC (ActUser, g_object_unref)
+
+static gboolean
+create_demo_user (GisDriver *driver)
+{
+  GisDriverPrivate *priv = gis_driver_get_instance_private (driver);
+  g_autoptr(ActUser) user;
+  GError *error = NULL;
+  const gchar *language;
+
+  error = NULL;
+  user = act_user_manager_create_user (act_user_manager_get_default (),
+                                       DEMO_ACCOUNT_USERNAME,
+                                       DEMO_ACCOUNT_FULLNAME,
+                                       ACT_USER_ACCOUNT_TYPE_ADMINISTRATOR,
```

Comment:
Ah yes, I meant to follow up on this and must have missed it. I'll have to test when using `ACT_USER_ACCOUNT_TYPE_STANDARD`. It *might* cause some problems in the sense that I think we pkexec some things from the demo-guest account, but testing will reveal whether that is actually the case.

For what its worth, this is the account type we use for the live user, though I think in that case, we actually want an administrator-level account since we could be installing from that session.